### PR TITLE
Add CHANGELOG for 1.6.0

### PR DESCRIPTION
**Pull request by beta-ziliani**
Preview: [changelog/1.6.0/CHANGELOG.md](https://github.com/beta-ziliani/crystal/blob/changelog/1.6.0/CHANGELOG.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,253 @@
...
+
+### Concurrency
+
+- Fix fiber clean loop on Windows ([#12300](https://github.com/crystal-lang/crystal/pull/12300), thanks @HertzDevil)
+- Enable `Mutex` on Windows ([#12213](https://github.com/crystal-lang/crystal/pull/12213), thanks @HertzDevil)
+
+### Crypto
+
+- Add support for Bcrypt algorithm version 2y ([#12447](https://github.com/crystal-lang/crystal/pull/12447), thanks @docelic)
+- Allow using U/Int128 in random ([#11977](https://github.com/crystal-lang/crystal/pull/11977), thanks @BlobCodes)
```
Comment (Sija) [inline CHANGELOG.md:33]:
```suggestion
- Allow using `U/Int128` in `Random` ([#11977](https://github.com/crystal-lang/crystal/pull/11977), thanks @BlobCodes)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,253 @@
...
+
+### Networking
+
+- Replace deprecated `Socket.ip?` with `IPAddress.valid?` ([#12489](https://github.com/crystal-lang/crystal/pull/12489), thanks @straight-shoota)
+- Fix `HTTP::Client#exec` to abort retry when client was closed ([#12465](https://github.com/crystal-lang/crystal/pull/12465), thanks @straight-shoota)
+- Fix specs with side effects ([#12539](https://github.com/crystal-lang/crystal/pull/12539), thanks @straight-shoota)
+- Fix `HTTP::Client` implicit compression with retry ([#12536](https://github.com/crystal-lang/crystal/pull/12536), thanks @straight-shoota)
+- `HTTP::StaticFileHandler`: Reduce max stat calls from 6 to 2 ([#12310](https://github.com/crystal-lang/crystal/pull/12310), thanks @didactic-drunk)
+- Add `Socket::IPAddress.valid?` ([#10492](https://github.com/crystal-lang/crystal/pull/10492), thanks @straight-shoota)
+- Add warning about concurrent requests in `HTTP::Client` ([#12527](https://github.com/crystal-lang/crystal/pull/12527), thanks @straight-shoota)
```
Comment (Sija) [inline CHANGELOG.md:61]:
```suggestion
- Fix `HTTP::Client#exec` to abort retry when client was closed ([#12465](https://github.com/crystal-lang/crystal/pull/12465), thanks @straight-shoota)
- Fix specs with side effects ([#12539](https://github.com/crystal-lang/crystal/pull/12539), thanks @straight-shoota)
- Fix `HTTP::Client` implicit compression with retry ([#12536](https://github.com/crystal-lang/crystal/pull/12536), thanks @straight-shoota)
- Reduce max stat calls from 6 to 2 in `HTTP::StaticFileHandler` ([#12310](https://github.com/crystal-lang/crystal/pull/12310), thanks @didactic-drunk)
- Add `Socket::IPAddress.valid?` ([#10492](https://github.com/crystal-lang/crystal/pull/10492), ([#12489](https://github.com/crystal-lang/crystal/pull/12489), thanks @straight-shoota)
- Add warning about concurrent requests in `HTTP::Client` ([#12527](https://github.com/crystal-lang/crystal/pull/12527), thanks @straight-shoota)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,253 @@
...
+- Make `sprintf` and `String#%` ignore NaN sign bit ([#12400](https://github.com/crystal-lang/crystal/pull/12400), thanks @HertzDevil)
+- Fix `Complex#to_s` imaginary component sign for certain values ([#12244](https://github.com/crystal-lang/crystal/pull/12244), thanks @HertzDevil)
+- More accurate definition of `Complex#sign` ([#12242](https://github.com/crystal-lang/crystal/pull/12242), thanks @HertzDevil)
+- Fix overflow for `rand(Range(Int, Int))` when signed span is too large ([#12545](https://github.com/crystal-lang/crystal/pull/12545), thanks @HertzDevil)
+- **(performance)** Add `#rotate_left` and `#rotate_right` for primitive integers ([#12307](https://github.com/crystal-lang/crystal/pull/12307), thanks @HertzDevil)
+- **(performance)** Optimize `BigDecimal#div` for inexact divisions ([#10803](https://github.com/crystal-lang/crystal/pull/10803), thanks @HertzDevil)
+- Implement the Dragonbox algorithm for `Float#to_s` ([#10913](https://github.com/crystal-lang/crystal/pull/10913), thanks @HertzDevil)
+- Add U/Int128 to `isqrt` spec ([#11976](https://github.com/crystal-lang/crystal/pull/11976), thanks @BlobCodes)
```
Comment (Sija) [inline CHANGELOG.md:74]:
```suggestion
- Add `U/Int128` to `isqrt` spec ([#11976](https://github.com/crystal-lang/crystal/pull/11976), thanks @BlobCodes)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,253 @@
...
+### Serialization
+
+- Fix YAML serialization class name ambiguity ([#12537](https://github.com/crystal-lang/crystal/pull/12537), thanks @hugopl)
+- Allow non-type converter instances in `ArrayConverter` and `HashValueConverter` ([#10638](https://github.com/crystal-lang/crystal/pull/10638), thanks @HertzDevil)
+
+### System
+
+- Add missing fields to `LibC::Passwd` on FreeBSD ([#12315](https://github.com/crystal-lang/crystal/pull/12315), thanks @dmgk)
+- Add platform-specific variants of `Process.parse_arguments` ([#12278](https://github.com/crystal-lang/crystal/pull/12278), thanks @HertzDevil)
+- Let `Dir.current` respect `$PWD` ([#12471](https://github.com/crystal-lang/crystal/pull/12471), thanks @straight-shoota)
```
Comment (Sija) [inline CHANGELOG.md:91]:
```suggestion
- Make `Dir.current` respect `$PWD` ([#12471](https://github.com/crystal-lang/crystal/pull/12471), thanks @straight-shoota)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,253 @@
...
+- **(performance)** Compiler: bind to tuple, not array ([#12423](https://github.com/crystal-lang/crystal/pull/12423), thanks @asterite)
+- Use `Path.new(string)` instead of `Path.new([string])` ([#12419](https://github.com/crystal-lang/crystal/pull/12419), thanks @asterite)
+- Decouple warning detection from program instances ([#12293](https://github.com/crystal-lang/crystal/pull/12293), thanks @HertzDevil)
+- **(performance)** Compiler: only have `freeze_type` in select AST nodes ([#12428](https://github.com/crystal-lang/crystal/pull/12428), thanks @asterite)
+- Correctly display codegen when cross-compiling ([#12414](https://github.com/crystal-lang/crystal/pull/12414), thanks @luislavena)
+- **(performance)** Compiler: avoid creating `Set` in `merge_if_vars` ([#12433](https://github.com/crystal-lang/crystal/pull/12433), thanks @asterite)
+- Compiler: simplify some calls ([#12417](https://github.com/crystal-lang/crystal/pull/12417), thanks @asterite)
+- **(performance)** Compiler: micro-optimizations in `merge_if_vars` ([#12432](https://github.com/crystal-lang/crystal/pull/12432), thanks @asterite)
```
Comment (Sija) [inline CHANGELOG.md:122]:
```suggestion
- **(performance)** Compiler: avoid creating `Set` in `merge_if_vars` ([#12433](https://github.com/crystal-lang/crystal/pull/12433), thanks @asterite)
- **(performance)** Compiler: micro-optimizations in `merge_if_vars` ([#12432](https://github.com/crystal-lang/crystal/pull/12432), thanks @asterite)
- Compiler: simplify some calls ([#12417](https://github.com/crystal-lang/crystal/pull/12417), thanks @asterite)
```

File: CHANGELOG.md (hunk 2 above)
Comment (straight-shoota) [reply CHANGELOG.md:61]: It's hard to figure out what's the actual change here...
Looks like combining #10492 and #12489 into a single entry? I guess that makes sense.

But please describe the description excplicitly. Just a diff may be okay for small typos, but such a reorganization is not easy to grok.

File: CHANGELOG.md (hunk 5 above)
Comment (straight-shoota) [reply CHANGELOG.md:122]: What's the reasoning here? It looks like just reordering the items?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,253 @@
...
+## Other
+
+- `bin/crystal`: Ensure `sh` compatibility ([#12486](https://github.com/crystal-lang/crystal/pull/12486), thanks @HertzDevil)
+- bumping version 1.6.0-dev ([#12263](https://github.com/crystal-lang/crystal/pull/12263), thanks @beta-ziliani)
+- updating CI to 1.5.0 ([#12260](https://github.com/crystal-lang/crystal/pull/12260), thanks @beta-ziliani)
+- Add fish shell completion ([#12026](https://github.com/crystal-lang/crystal/pull/12026), thanks @TunkShif)
+- Execute `compopt` only when it's present ([#12248](https://github.com/crystal-lang/crystal/pull/12248), thanks @potomak)
+- Use `Makefile.win` and wrapper script on Windows CI ([#12344](https://github.com/crystal-lang/crystal/pull/12344), thanks @HertzDevil)
+- [Makefile] Add format target ([#11420](https://github.com/crystal-lang/crystal/pull/11420), thanks @straight-shoota)
+- Spec: bump and document timeouts in interpreted mode ([#12430](https://github.com/crystal-lang/crystal/pull/12430), thanks @asterite)
```
Comment (straight-shoota) [inline CHANGELOG.md:238]: This PR was missing labels. It belongs into stdlib section.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,253 @@
...
+- Merge release 1.5.1 into master ([#12460](https://github.com/crystal-lang/crystal/pull/12460), thanks @straight-shoota)
```
Comment (straight-shoota) [inline CHANGELOG.md:239]: This is just a merge commit, not relevant for the changelog.
```suggestion
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,253 @@
...
+- Update contact section of CODE of CONDUCT ([#9219](https://github.com/crystal-lang/crystal/pull/9219), thanks @paulcsmith)
+- Update nixpkgs 22.05 and LLVM 11 ([#12498](https://github.com/crystal-lang/crystal/pull/12498), thanks @straight-shoota)
+- [Makefile] Use `EXPORT_CC` for `make crystal` ([#11760](https://github.com/crystal-lang/crystal/pull/11760), thanks @straight-shoota)
+- Update distribution-scripts ([#12502](https://github.com/crystal-lang/crystal/pull/12502), thanks @straight-shoota)
+- Fix and enhance `scripts/update-distribution-scripts.sh` ([#12503](https://github.com/crystal-lang/crystal/pull/12503), thanks @straight-shoota)
+- [CI] Upgrade GitHub Actions to macos-11 ([#12500](https://github.com/crystal-lang/crystal/pull/12500), thanks @straight-shoota)
+- Add icon and metadata to Windows Crystal compiler binary ([#12494](https://github.com/crystal-lang/crystal/pull/12494), thanks @HertzDevil)
+- Remove `spec/win32_std_spec.cr` ([#12282](https://github.com/crystal-lang/crystal/pull/12282), thanks @HertzDevil)
+- Remove `spec/generate_windows_spec.sh` ([#12549](https://github.com/crystal-lang/crystal/pull/12549), thanks @straight-shoota)
```
Comment (straight-shoota) [inline CHANGELOG.md:248]: I think these could be merged into one item:
```suggestion
- Remove `spec/win32_std_spec.cr` ([#12282](https://github.com/crystal-lang/crystal/pull/12282), ([#12549](https://github.com/crystal-lang/crystal/pull/12549) thanks @HertzDevil, @straight-shoota)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,253 @@
...
+- Update distribution-scripts ([#12555](https://github.com/crystal-lang/crystal/pull/12555), thanks @straight-shoota)
```
Comment (straight-shoota) [inline CHANGELOG.md:249]: This could also be merged with the other *Update distribution-scripts* item.

File: CHANGELOG.md (hunk 5 above)
Comment (Sija) [reply CHANGELOG.md:122]: yeah, seemed odd to have two entries related to the same part of the codebase separated.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,254 @@
...
+- **(performance)** Introduce `IO::DEFAULT_BUFFER_SIZE` ([#12507](https://github.com/crystal-lang/crystal/pull/12507), thanks @straight-shoota)
+- Add support for `IO::FileDescriptor` staying open on finalize ([#12367](https://github.com/crystal-lang/crystal/pull/12367), thanks @refi64)
+
+### Macros
+
+- Enhance `record` documentation ([#12334](https://github.com/crystal-lang/crystal/pull/12334), thanks @straight-shoota)
+
+### Networking
+
+- Replace deprecated `Socket.ip?` with `IPAddress.valid?` ([#12489](https://github.com/crystal-lang/crystal/pull/12489), [#10492](https://github.com/crystal-lang/crystal/pull/10492), thanks @straight-shoota)
```
Comment (Sija) [inline CHANGELOG.md:55]:
```suggestion
- Add `Socket::IPAddress.valid?` ([#12489](https://github.com/crystal-lang/crystal/pull/12489), [#10492](https://github.com/crystal-lang/crystal/pull/10492), thanks @straight-shoota)
```

**Review by straight-shoota — APPROVED** [review]
I'm approving this to make it clear for merging when ready.

But it looks like https://github.com/crystal-lang/crystal/pull/12568/files#r988795474, https://github.com/crystal-lang/crystal/pull/12568#discussion_r988990167 and https://github.com/crystal-lang/crystal/pull/12568#discussion_r989136883 still need addressing.

File: CHANGELOG.md (hunk 5 above)
Comment (beta-ziliani) [reply CHANGELOG.md:122]: I merged them

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**